website/homepage/src/demos/restart_demos.py

```diff
@@ -0,0 +1,10 @@
+from upload_demos import demos_by_category, upload_demo_to_space, AUTH_TOKEN, gradio_version
+from gradio.networking import url_ok
+
+
+for category in demos_by_category:
+    for demo in category["demos"]:
+        space_id = "gradio/" + demo["dir"]
+        if not url_ok(f"https://hf.space/embed/{space_id}/+"):
+            print(f"{space_id} was down, restarting")
+            upload_demo_to_space(demo_name=demo["dir"], space_id=space_id, hf_token=AUTH_TOKEN, gradio_version=gradio_version)
```

Comment:
Will this trigger a restart if this files haven't changed?

Reply:
Yes, which is intended. We don't only restart demos when files have changed. We also restart them when the version is changed. This just restarts them if they're down.

Reply:
Agreed. My question is, will this be *sufficient* to restart the Space? In other words, if you push to a Space and the files haven't changed, will it actually restart the Space?

Reply:
Agreed. My question is, will this be *sufficient* to restart the Space? In other words, if you push to a Space and the files haven't changed, will it actually restart the Space?

Reply:
Oh got it @abidlabs, yes it will.